test/framework/src/main/java/org/opensearch/test/OpenSearchIntegTestCase.java

```diff
@@ -2079,9 +2079,13 @@ protected boolean addMockGeoShapeFieldMapper() {
         return true;
     }
 
-    /** Returns {@code true} if this test cluster should have tracing enabled with MockTelemetryPlugin */
+    /**
+     * Returns {@code true} if this test cluster should have tracing enabled with MockTelemetryPlugin
+     * Disabling this for now as the existing way of strict check do not support multiple nodes internal cluster.
+     * @return boolean.
+     */
     protected boolean addMockTelemetryPlugin() {
-        return true;
+        return false;
```

Comment:
This is concerning, what issues do we have with testing over Mock plugin?

Reply:
But the nodes are getting closed in loop one after another.

Reply:
You mean that the tests is trying to validate before the nodes shuts down? (so there may be still some traces coming in)

Reply:
Yes.

Reply:
So ... I enabled the mock tracer & span verification, to my surprise, the single node test case `org.opensearch.cluster.routing.ShardMovementStrategyTests` failed and revealed that we don't handle errors properly:


```
java.lang.AssertionError: SpanData validation failed for validator org.opensearch.test.telemetry.tracing.validators.AllSpansAreEndedProperly/nMockSpanData{spanID='7325a527aee1d305', parentSpanID='', traceID='ffebfd5781d9377c', spanName='internal:cluster/coordination/publish_state 127.0.0.1', startEpochNanos=37488469003160, endEpochNanos=0, hasEnded=false, stackTrace=[java.base/java.lang.Thread.getStackTrace(Thread.java:2514), org.opensearch.test.telemetry.tracing.StrictCheckSpanProcessor.toMockSpanData(StrictCheckSpanProcessor.java:60), org.opensearch.test.telemetry.tracing.StrictCheckSpanProcessor.onStart(StrictCheckSpanProcessor.java:31), org.opensearch.test.telemetry.tracing.MockTracingTelemetry.createSpan(MockTracingTelemetry.java:48), org.opensearch.telemetry.tracing.DefaultTracer.createSpan(DefaultTracer.java:106), org.opensearch.telemetry.tracing.DefaultTracer.startSpan(DefaultTracer.java:66), org.opensearch.telemetry.tracing.WrappedTracer.startSpan(WrappedTracer.java:81), org.opensearch.telemetry.tracing.WrappedTracer.startSpan(WrappedTracer.java:54), org.opensearch.telemetry.tracing.WrappedTracer.startSpan(WrappedTracer.java:44), org.opensearch.transport.TransportService.sendRequest(TransportService.java:870), org.opensearch.transport.TransportService.sendRequest(TransportService.java:848), org.opensearch.cluster.coordination.PublicationTransportHandler$PublicationContext.sendClusterState(PublicationTransportHandler.java:455), org.opensearch.cluster.coordination.PublicationTransportHandler$PublicationContext.sendClusterStateDiff(PublicationTransportHandler.java:412), org.opensearch.cluster.coordination.PublicationTransportHandler$PublicationContext.sendPublishRequest(PublicationTransportHandler.java:347), org.opensearch.cluster.coordination.Coordinator$CoordinatorPublication.sendPublishRequest(Coordinator.java:1800), org.opensearch.cluster.coordination.Publication$PublicationTarget.sendPublishRequest(Publication.java:287), java.base/java.util.ArrayList.forEach(ArrayList.java:1511), org.opensearch.cluster.coordination.Publication.start(Publication.java:94), org.opensearch.cluster.coordination.Coordinator.publish(Coordinator.java:1314), org.opensearch.cluster.service.MasterService.publish(MasterService.java:349), org.opensearch.cluster.service.MasterService.runTasks(MasterService.java:331), org.opensearch.cluster.service.MasterService$Batcher.run(MasterService.java:206), org.opensearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:204), org.opensearch.cluster.service.TaskBatcher$BatchedTask.run(TaskBatcher.java:242), org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:849), org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedOpenSearchThreadPoolExecutor.java:282), org.opensearch.common.util.concurrent.PrioritizedOpenSearchThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedOpenSearchThreadPoolExecutor.java:245), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642), java.base/java.lang.Thread.run(Thread.java:1623)]}
	at __randomizedtesting.SeedInfo.seed([6F5BC9E83BEEDA04:1DF6FAF4DD97305E]:0)
	at org.opensearch.test.telemetry.tracing.TelemetryValidators.validate(TelemetryValidators.java:43)
	at org.opensearch.test.telemetry.tracing.MockTracingTelemetry.close(MockTracingTelemetry.java:67)
	at org.opensearch.telemetry.tracing.DefaultTracer.close(DefaultTracer.java:75)
	at org.opensearch.telemetry.tracing.WrappedTracer.close(WrappedTracer.java:86)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:89)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:131)
	at org.opensearch.common.util.io.IOUtils.close(IOUtils.java:114)
	at org.opensearch.node.Node.close(Node.java:1552)
	at org.opensearch.test.InternalTestCluster$NodeAndClient.close(InternalTestCluster.java:1127)
	at org.opensearch.test.InternalTestCluster.stopNodesAndClients(InternalTestCluster.java:1918)
	at org.opensearch.test.InternalTestCluster.stopNodesAndClient(InternalTestCluster.java:1908)
	at org.opensearch.test.InternalTestCluster.stopRandomNode(InternalTestCluster.java:1792)
	at org.opensearch.cluster.routing.ShardMovementStrategyTests.testClusterGreenAfterPartialRelocation(ShardMovementStrategyTests.java:164)
	at org.opensearch.cluster.routing.ShardMovementStrategyTests.testClusterRelocationNoPreferenceShardMovementPrimaryFirstEnabled(ShardMovementStrategyTests.java:75)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
```

Reply:
The issue should be fixed now, we could revert the change, right?

Reply:
yes, looks like merge related issue. Fixing it.